themes/divi-child-bulle/functions.php

```diff
@@ -565,6 +578,23 @@ function dismoi_get_informateur_id( ) {
     return null;
 }
 
+/**
+ * Get Informateur by id
+ *
+ * @return stdClass|null
+ */
+function dismoi_get_informateur_by_id( ) {
+    $id = dismoi_get_informateur_id();
+    if (!empty($id)) {
+        if (!empty( $profile_object ) && count( $profile_object ) > 0)  {
```

Comment:
I think there's a problem of scope here with $profile_object. We'd need to declare it as global if we want to refer to a top level variable. We need to use get_profile_object here I think. I'll make the changes
```
function dismoi_get_informateur_by_id( ) {
    $id = dismoi_get_informateur_id();
    if (!empty($id)) {
        return get_profile_object( $id );
    }

    return null;
}
```

Reply:
@graemehoffman Yeah my bad. I put that in the wrong place.
The idea is to put these 2 tests:
```
!empty( $profile_object ) && count( $profile_object ) > 0
```

**In** the `get_profile_object` function and return null if they don't apply. Thus you won't need these tests in the other functions: you'll just have to test the whole object `if($profile_object) { //...`